feat: Add docs for *VideoFrameCallback() methods (#31762)
mdn/content: files/en-us/web/api/htmlvideoelement/cancelvideoframecallback/index.md

---
title: "HTMLVideoElement: cancelVideoFrameCallback() method"
short-title: cancelVideoFrameCallback()
slug: Web/API/HTMLVideoElement/cancelVideoFrameCallback
page-type: web-api-instance-method
browser-compat: api.HTMLVideoElement.cancelVideoFrameCallback
---

> {{APIRef("Picture-in-Picture API")}}

Reviewer: I think it is not good to use {{APIRef("Picture-in-Picture API")}}, since it is not a part of the Picture-in-Picture API. And it is defined in a different spec, https://wicg.github.io/video-rvfc/. If the API summary will not be created, a {{APIRef("HTML DOM")}} might be better

Author: Argh, that was a copy/paste error! I originally used the `requestPictureInPicture()` page as a template for this one.

Changed to "HTML DOM" in my next commit ;-)

> The **`cancelVideoFrameCallback()`** method of the {{domxref("HTMLVideoElement")}} interface cancels a previously-registered video frame request callback.

Reviewer 2: ```suggestion
The **`cancelVideoFrameCallback()`** method of the {{domxref("HTMLVideoElement")}} interface cancels a previously-registered video frame callback.
```

## Syntax

```js-nolint
cancelVideoFrameCallback(id)
```

### Parameters

- `id`
  - : A number representing the ID of the video frame callback you want to cancel. This will be the value returned by the corresponding {{DOMxRef("HTMLVideoElement.requestVideoFrameCallback")}} call.

### Return value

> `undefined`.

Reviewer: ```suggestion
None ({{jsxref('undefined')}}).
```

same as the [API method template](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_method_subpage_template)

Author: Updated in next commit

## Examples

### Cancel a video frame callback

This example shows how to use `cancelVideoFrameCallback()` to cancel a previously registered video frame callback.

```js
let paintCount = 0;
let startTime = 0.0;

const updateCanvas = (now, metadata) => {
  // Do something with the frame.

  // ...

  // Re-register the callback to be notified about the next frame.

>   video.requestVideoFrameCallback(updateCanvas);
> };
> 
> // Initially register the callback to be notified about the first frame.
> const videoCallbackId = video.requestVideoFrameCallback(updateCanvas);

Reviewer 2: ```suggestion
    // It's important to update the videoCallbackId each time you're calling requestVideoFrameCallback.
    videoCallbackId = video.requestVideoFrameCallback(updateCanvas);
};

// Initially register the callback to be notified about the first frame.
let videoCallbackId = video.requestVideoFrameCallback(updateCanvas);
```

Reviewer 2: Also see https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame#examples:~:text=//%20it%27s%20important%20to%20update%20the%20requestId%20each%20time%20you%27re%20calling%20requestAnimationFrame

Author: Ah, makes sense. I've updated the example as suggested but cleaned it up a bit and added some extra commentary to make it clearer why you need to do this.

---
title: "HTMLVideoElement: cancelVideoFrameCallback() method"
short-title: cancelVideoFrameCallback()
slug: Web/API/HTMLVideoElement/cancelVideoFrameCallback
page-type: web-api-instance-method
browser-compat: api.HTMLVideoElement.cancelVideoFrameCallback
---

{{APIRef("HTML DOM")}}

The **`cancelVideoFrameCallback()`** method of the {{domxref("HTMLVideoElement")}} interface cancels a previously-registered video frame callback.

## Syntax

```js-nolint
cancelVideoFrameCallback(id)
```

### Parameters

- `id`
  - : A number representing the ID of the video frame callback you want to cancel. This will be the value returned by the corresponding {{DOMxRef("HTMLVideoElement.requestVideoFrameCallback")}} call.

### Return value

None ({{jsxref("undefined")}}).

## Examples

> ### Cancel a video frame callback

Author: Updated!

> This example shows how to use `cancelVideoFrameCallback()` to cancel a previously registered video frame callback.

Author: Updated

```js
const updateCanvas = (now, metadata) => {
  // Do something with the frame.

  // ...

  // Re-register the callback to be notified about the next frame.
  // It's important to update the videoCallbackId on each iteration

>   // so you can cancel the callback successfully

Author: I've instead just removed the periods from the others. It feels weird to put periods at the end of code comments.

  videoCallbackId = video.requestVideoFrameCallback(updateCanvas);
};

> // Initially register the callback to be notified about the first frame.

Author: Updated these comments in the same way as explained earlier.
